packages/core/src/common/keybinding.ts

```diff
@@ -267,4 +334,42 @@ export class KeybindingRegistry {
     isPseudoCommand(commandId: string): boolean {
         return commandId === KeybindingRegistry.PASSTHROUGH_PSEUDO_COMMAND;
     }
+
+    setKeymap(rawKeyBindings: RawKeybinding[]) {
+        const customBindings: Keybinding[] = [];
+        for (const rawKeyBinding of rawKeyBindings) {
+            if (this.commandRegistry.getCommand(rawKeyBinding.command)) {
+                // TODO check if valid context? valid keybinding?
+                // Create Keycode from string?
+                // Create new Keybinding and assign it
+                // If one keybinding is wrong, reset to the default ones.
+
+                const code = KeyCode.parseKeystroke(rawKeyBinding.keybinding);
```

Comment:
This should be KeyCode.parse(string) since it returns a KeyCode... the documentation is also wrong as:

```typescript
    /**
     * Parses a Keystroke object from a string.
     * @param keybinding String representation of a keybinding
     */
    public static parseKeystroke(keybinding: string): KeyCode | undefined {
```
This creates a KeyCode.. there's no such thing as a Keystroke object